### ISSUE TITLE
Problem custom dataset

### ISSUE THREAD

**Issue by robbe000** [KaiyangZhou/deep-person-reid#326]
I tried to train on a custom dataset, but I almost immediately get 100% acc. Do you have any idea how to solve this problem? I attached my code and output.

Code:

    class NewDataset(ImageDataset):
         dataset_dir = 'koe_dataset1'

        def __init__(self, root='', **kwargs):
            self.root = osp.abspath(osp.expanduser(root))
            self.dataset_dir = osp.join(self.root, self.dataset_dir)
            print(self.dataset_dir)

            train = gegevens[0]
            query = gegevens[1]
            gallery = gegevens[2]

        super(NewDataset, self).__init__(train, query, gallery, **kwargs)

    torchreid.data.register_image_dataset('koe_dataset1', NewDataset)

    datamanager = torchreid.data.ImageDataManager(
        root='reid-data',
        sources='koe_dataset1',
        height=256,
        width=128,
        batch_size_train=32,
        batch_size_test=32,
    )

    model = torchreid.models.build_model(
        name='resnet50',
        num_classes=datamanager.num_train_pids,
        loss='softmax'
    )

    model = model.cuda()

    optimizer = torchreid.optim.build_optimizer(
        model,
        optim='adam',
        lr=0.0003
    )

    scheduler = torchreid.optim.build_lr_scheduler(
        optimizer,
        lr_scheduler='single_step',
        stepsize=20
    )

    engine = torchreid.engine.ImageSoftmaxEngine(
        datamanager,
        model,
        optimizer=optimizer,
        scheduler=scheduler
    )

    engine.run(
        save_dir='log/resnet50',
        max_epoch=60,
        eval_freq=10,
        print_freq=10
    )

Output:

Epoch: [1/60][10/16]	Time 0.208 (0.267)	Data 0.000 (0.037)	Loss 1.4263 (2.0978)	Acc 68.75 (50.62)	Lr 0.000300	eta 0:04:13
Epoch: [2/60][10/16]	Time 0.210 (0.261)	Data 0.000 (0.048)	Loss 0.7205 (0.8085)	Acc 100.00 (96.88)	Lr 0.000300	eta 0:04:03
Epoch: [3/60][10/16]	Time 0.209 (0.257)	Data 0.000 (0.046)	Loss 0.7382 (0.7110)	Acc 100.00 (98.75)	Lr 0.000300	eta 0:03:55
Epoch: [4/60][10/16]	Time 0.210 (0.257)	Data 0.000 (0.048)	Loss 0.6940 (0.6678)	Acc 100.00 (100.00)	Lr 0.000300	eta 0:03:52
Epoch: [5/60][10/16]	Time 0.206 (0.258)	Data 0.000 (0.047)	Loss 0.6917 (0.6698)	Acc 96.88 (98.75)	Lr 0.000300	eta 0:03:49
Epoch: [6/60][10/16]	Time 0.210 (0.258)	Data 0.000 (0.046)	Loss 0.6607 (0.6573)	Acc 96.88 (99.06)	Lr 0.000300	eta 0:03:44
Epoch: [7/60][10/16]	Time 0.215 (0.259)	Data 0.000 (0.043)	Loss 0.6445 (0.6453)	Acc 100.00 (100.00)	Lr 0.000300	eta 0:03:40
Epoch: [8/60][10/16]	Time 0.211 (0.261)	Data 0.000 (0.047)	Loss 0.6237 (0.6353)	Acc 100.00 (100.00)	Lr 0.000300	eta 0:03:38
Epoch: [9/60][10/16]	Time 0.214 (0.258)	Data 0.000 (0.043)	Loss 0.6282 (0.6259)	Acc 100.00 (100.00)	Lr 0.000300	eta 0:03:31
Epoch: [10/60][10/16]	Time 0.213 (0.260)	Data 0.000 (0.045)	Loss 0.6304 (0.6179)	Acc 100.00 (100.00)	Lr 0.000300	eta 0:03:29
##### Evaluating koe_dataset1 (source) #####
Extracting features from query set ...
Done, obtained 133-by-2048 matrix
Extracting features from gallery set ...
Done, obtained 135-by-2048 matrix
Speed: 0.0099 sec/batch
Computing distance matrix with metric=euclidean ...
Computing CMC and mAP ...
** Results **
mAP: 99.6%
CMC curve
Rank-1  : 100.0%
Rank-5  : 100.0%
Rank-10 : 100.0%
Rank-20 : 100.0%
Checkpoint saved to "log/resnet50/model.pth.tar-10"
Epoch: [11/60][10/16]	Time 0.211 (0.273)	Data 0.000 (0.059)	Loss 0.6173 (0.6153)	Acc 100.00 (100.00)	Lr 0.000300	eta 0:03:35
Epoch: [12/60][10/16]	Time 0.217 (0.260)	Data 0.000 (0.045)	Loss 0.6124 (0.6144)	Acc 100.00 (100.00)	Lr 0.000300	eta 0:03:21
Epoch: [13/60][10/16]	Time 0.214 (0.263)	Data 0.000 (0.045)	Loss 0.6054 (0.6067)	Acc 100.00 (100.00)	Lr 0.000300	eta 0:03:19
Epoch: [14/60][10/16]	Time 0.216 (0.251)	Data 0.000 (0.029)	Loss 0.6058 (0.6076)	Acc 100.00 (100.00)	Lr 0.000300	eta 0:03:05

**Comment by KaiyangZhou** [comment]
have you checked if `train` overlaps with `query` and `gallery`?

what's the task you are solving? is it too easy?

**Comment by robbe000** [comment]
I am trying to reidentify cows. I don't think that the task is to easy because I tested the model and it doesn't seem to work. I attached a part of the train, query and gallery list.

```
  ----------------------------------------
  subset   | # ids | # images | # cameras
  ----------------------------------------
  train    |    20 |      513 |         2
  query    |    20 |      133 |         2
  gallery  |    20 |      135 |         2
  ----------------------------------------
```

Train:
`[('/content/deep-person-reid/reid-data/koe_dataset1/output2/1_0.jpg', 0, 0), ('/content/deep-person-reid/reid-data/koe_dataset1/output2/1_114.jpg', 0, 0), ('/content/deep-person-reid/reid-data/koe_dataset1/output2/1_152.jpg', 0, 0), ('/content/deep-person-reid/reid-data/koe_dataset1/output2/1_190.jpg', 0, 0), ('/content/deep-person-reid/reid-data/koe_dataset1/output2/1_228.jpg', 0, 0), ('/content/deep-person-reid/reid-data/koe_dataset1/output2/2_0.jpg', 1, 0), ('/content/deep-person-reid/reid-data/koe_dataset1/output2/2_3.jpg', 1, 0), ('/content/deep-person-reid/reid-data/koe_dataset1/output2/2_4.jpg', 1, 0), ('/content/deep-person-reid/reid-data/koe_dataset1/output2/2_5.jpg', 1, 0), ('/content/deep-person-reid/reid-data/koe_dataset1/output2/2_6.jpg', 1, 0), ('/content/deep-person-reid/reid-data/koe_dataset1/output2/2_9.jpg', 1, 0), ('/content/deep-person-reid/reid-data/koe_dataset1/output2/2_10.jpg', 1, 0), ('/content/deep-person-reid/reid-data/koe_dataset1/output2/2_11.jpg', 1, 0), ('/content/deep-person-reid/reid-data/koe_dataset1/output2/2_12.jpg', 1, 0),...`

Query:
`[('/content/deep-person-reid/reid-data/koe_dataset1/output2/1_76.jpg', 0, 0), ('/content/deep-person-reid/reid-data/koe_dataset1/output2/1_304.jpg', 0, 0), ('/content/deep-person-reid/reid-data/koe_dataset1/output2/2_2.jpg', 1, 0), ('/content/deep-person-reid/reid-data/koe_dataset1/output2/2_8.jpg', 1, 0), ('/content/deep-person-reid/reid-data/koe_dataset1/output2/2_14.jpg', 1, 0), ('/content/deep-person-reid/reid-data/koe_dataset1/output2/2_20.jpg', 1, 0), ('/content/deep-person-reid/reid-data/koe_dataset1/output2/2_26.jpg', 1, 0), ('/content/deep-person-reid/reid-data/koe_dataset1/output2/2_32.jpg', 1, 0), ('/content/deep-person-reid/reid-data/koe_dataset1/output2/2_38.jpg', 1, 0), ('/content/deep-person-reid/reid-data/koe_dataset1/output2/3_2.jpg', 2, 0), ('/content/deep-person-reid/reid-data/koe_dataset1/output2/3_8.jpg', 2, 0), ('/content/deep-person-reid/reid-data/koe_dataset1/output2/3_14.jpg', 2, 0),...`

Gallery:
`[('/content/deep-person-reid/reid-data/koe_dataset1/output2/1_38.jpg', 0, 1), ('/content/deep-person-reid/reid-data/koe_dataset1/output2/1_266.jpg', 0, 1), ('/content/deep-person-reid/reid-data/koe_dataset1/output2/2_1.jpg', 1, 1), ('/content/deep-person-reid/reid-data/koe_dataset1/output2/2_7.jpg', 1, 1), ('/content/deep-person-reid/reid-data/koe_dataset1/output2/2_13.jpg', 1, 1), ('/content/deep-person-reid/reid-data/koe_dataset1/output2/2_19.jpg', 1, 1), ('/content/deep-person-reid/reid-data/koe_dataset1/output2/2_25.jpg', 1, 1), ('/content/deep-person-reid/reid-data/koe_dataset1/output2/2_31.jpg', 1, 1),...`

**Comment by KaiyangZhou** [comment]
try visualizing the ranking results to see if the images really match https://kaiyangzhou.github.io/deep-person-reid/user_guide.html#visualize-ranking-results
